src/code/m_controller.c

```diff
@@ -0,0 +1,114 @@
+#include "m_controller.h"
+#include "game.h"
+#include "m_event.h"
+#include "m_lib.h"
+#include "sys_math_atan.h"
+
+void mCon_ct(GameState* arg0) {
+
+}
+
+void mCon_dt(GameState* arg0) {
+
+}
+
+void mCon_calc(Controller* controller, f32 x, f32 y) {
+    controller->lastMoveX = controller->moveX;
+    controller->lastMoveY = controller->moveY;
+    controller->lastMoveR = controller->moveR;
+
+    controller->lastMoveAngle = controller->moveAngle;
+
+    controller->lastAdjustedX = controller->adjustedX;
+    controller->lastAdjustedY = controller->adjustedY;
+    controller->lastAdjustedR = controller->adjustedR;
+
+    {
+        f32 r = sqrtf((x * x) + (y * y));
+
+        if (r <= STICK_MIN) {
+            controller->moveX = 0.0f;
+            controller->moveY = 0.0f;
+            controller->moveR = 0.0f;
+            controller->adjustedX = 0.0f;
+            controller->adjustedY = 0.0f;
+            controller->adjustedR = 0.0f;
+        } else {
+            s16 theta = atans_table(x, y);
+
+            if (r > STICK_MAX) {
+                x = cos_s(theta) * STICK_MAX;
+                y = sin_s(theta) * STICK_MAX;
+                r = STICK_MAX;
+            }
+
+            controller->moveAngle = theta;
+            {
+                s32 temp;
+            }
+            controller->moveX = check_percent_abs(x, STICK_MIN, STICK_MAX, STICK_UNCORRECTED_SCALE, 0);
+            controller->moveY = check_percent_abs(y, STICK_MIN, STICK_MAX, STICK_UNCORRECTED_SCALE, 0);
+            controller->moveR = check_percent_abs(r, STICK_MIN, STICK_MAX, STICK_UNCORRECTED_SCALE, 0);
+
+            controller->adjustedX = check_percent_abs(x, STICK_MIN, STICK_MAX, STICK_CORRECTED_SCALE, 1);
+            controller->adjustedY = check_percent_abs(y, STICK_MIN, STICK_MAX, STICK_CORRECTED_SCALE, 1);
+            controller->adjustedR = check_percent_abs(r, STICK_MIN, STICK_MAX, STICK_CORRECTED_SCALE, 1);
+        }
+    }
+}
+
+void mCon_main(GameState* game) {
+    Controller* controller = &game->controller;
+    f32 x = getJoystick_X();
+    f32 y = getJoystick_Y();
+
+    mCon_calc(controller, x, y);
+}
+
+s32 chkButton(u16 mask) {
+    if (mEv_CheckTitleDemo() > 0) {
+        return 0;
+    } else {
+        return mask == (gamePT->input[0].cur.button & mask);
```

Comment:
There's the `CONTROLLER1` macro in `game.h`, I think it would be better to use that instead

Reply:
It doesn't match here apparently? matches in the other functions but can't get to match it here with that macro

Reply:
nvm me dumb lol